### PR TITLE
[DEV-1110] Add custom-message-forgot-password e-mail template

### PR DESCRIPTION
**Pull request by marcobottaro**
#### List of Changes

Add custom-message-forgot-password e-mail template

#### How Has This Been Tested?

Adding console.log(emailMessage) after custom-message-forgot-password-handler.ts:23 in order to print emailMessage output in console and running tests with this command:
npm -w cognito-functions run test -- --watch 

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/cognito-functions/src/__tests__/custom-message-forgot-password-handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  userName: 'aUserName',
+  callerContext: {
+    awsSdkVersion: 'anAwsSdkVersion',
+    clientId: 'aClientId',
+  },
+  request: {
+    userAttributes: {
+      sub: 'user-identity',
+    },
+    codeParameter: '####',
```
Comment (marcobottaro) [inline apps/cognito-functions/src/__tests__/custom-message-forgot-password-handler.test.ts:19]: Check this

File: apps/cognito-functions/src/__tests__/custom-message-forgot-password-handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('Handler', () => {
+  const env = {
+    domain: 'thedomain.org',
+  };
+  it('should reply with forgot password link', async () => {
+    const { response } = await makeHandler(env)(event);
+    const { userAttributes } = event.request;
+    const expected = makeConfirmationForgotPasswordEmail(
+      `https://${env.domain}/auth/forgot-password?username=${userAttributes['sub']}`
```
Comment (marcobottaro) [inline apps/cognito-functions/src/__tests__/custom-message-forgot-password-handler.test.ts:38]: Check URL

File: apps/cognito-functions/src/custom-message-forgot-password-handler.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  typeof CustomMessageForgotPasswordEnv
+>;
+
+export const makeHandler =
+  (env: CustomMessageForgotPasswordEnv) =>
+  async (event: CustomMessageForgotPasswordTriggerEvent) => {
+    const username = event.request.userAttributes['sub'];
+
+    if (event.triggerSource === 'CustomMessage_ForgotPassword') {
+      // TODO: check if codeParameter is necessary
```
Comment (marcobottaro) [inline apps/cognito-functions/src/custom-message-forgot-password-handler.ts:18]: Check

File: apps/cognito-functions/src/custom-message-forgot-password-handler.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      // const { codeParameter } = event.request;
+      // &code=${codeParameter}`;
+      // TODO: check url
```
Comment (marcobottaro) [inline apps/cognito-functions/src/custom-message-forgot-password-handler.ts:21]: Check

**Comment by kin0992** [comment]
@marcobottaro please remove all the stuff **NOT** related to the email template.
The handler will be done in a separate PR. 🙏

**Comment by datalek** [comment]
Replaced by #401

**Pull request closed without merging**